false-positive: reassign inside an infinite loop and not use after exiting loop

```go
package main

func main() {
	var err error
	var rest []byte
	for {
		rest, err = bar(rest)

		if err == nil {
			break
		}
	}
}

func bar(d []byte) ([]byte, error) {
	return nil, nil
}
```

```
foo.go:7:3: reassigned, but never used afterwards (wastedassign)
                rest, err = bar(rest)
                ^
```

--- a/testdata/src/a/a.go
+++ b/testdata/src/a/a.go
@@ -109,7 +109,7 @@ func mugen() {
 	var i int
 	var hoge int
 	for {
-		hoge = 5 // want "reassigned, but never used afterwards"
+		hoge = 5 // want "wasted assignment"
 		// break
 	}
 
@@ -129,4 +129,27 @@ func noMugen() {
 	println(i)
 	println(hoge)
 	return
+}
+
+func reassignInsideLoop() {
+	bar := func(b []byte) ([]byte, error) { return b, nil }
+	var err error
+	var rest []byte
+	for {
+		rest, err = bar(rest)
+		if err == nil {
+			break
+		}
+	}
+	return
+}
+
+func reassignInsideLoop2() {
+	var x int = 0
+	var y int = 1
+	for i := 1; i < 3; i++ {
+		x += y
+		y *= 2 * i
+	}
+	println(x)
 }

--- a/wastedassign.go
+++ b/wastedassign.go
@@ -163,20 +163,20 @@ func (wr wastedReason) String() string {
 	}
 }
 
-func isNextOperationToOpIsStore(bls []*ssa.BasicBlock, currentOp *ssa.Value, haveCheckedMap map[int]bool) wastedReason {
+func isNextOperationToOpIsStore(bls []*ssa.BasicBlock, currentOp *ssa.Value, haveCheckedMap map[int]int) wastedReason {
 	var wastedReasons []wastedReason
 	var wastedReasonsCurrentBls []wastedReason
 
 	if haveCheckedMap == nil {
-		haveCheckedMap = map[int]bool{}
+		haveCheckedMap = map[int]int{}
 	}
 
 	for _, bl := range bls {
-		if haveCheckedMap[bl.Index] {
+		if haveCheckedMap[bl.Index] == 2 {
 			continue
 		}
 
-		haveCheckedMap[bl.Index] = true
+		haveCheckedMap[bl.Index]++
 		breakFlag := false
 		for _, ist := range bl.Instrs {
 			if breakFlag {
@@ -201,7 +201,7 @@ func isNextOperationToOpIsStore(bls []*ssa.BasicBlock, currentOp *ssa.Value, hav
 				var buf [10]*ssa.Value
 				for _, op := range ist.Operands(buf[:0]) {
 					if *op == *currentOp {
-						// 連続storeではなかった
+						// It wasn't a continuous store.
 						return notWasted
 					}
 				}

